### PR TITLE
Add nodev,noexec,nosuid options to /var/log and /var/log/audit.

### PR DESCRIPTION
**Pull request by adelton**
#### Description:

- Add nodev,noexec,nosuid options to /var/log and /var/log/audit.

#### Rationale:

- Restricting operations which should never be needed on /var/log/*.

### CODE REVIEW

**Comment by adelton** [comment]
Similar to https://github.com/ComplianceAsCode/content/pull/4542, we'd need to know the severity, the references, whether to list older RHEL versions, and if to add this to the ospp.profile files right away.

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_noexec/rule.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+title: 'Add noexec Option to /var/log'
+
+description: |-
+    The <tt>noexec</tt> mount option can be used to prevent binaries
+    from being executed out of <tt>/var/log</tt>.
+    {{{ describe_mount(option="noexec", part="/var/log") }}}
+
+rationale: |-
+    Allowing users to execute binaries from world-writable directories
```
Comment (yuumasato) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_noexec/rule.yml]: Are `/var` and  `/var/log` world-writable?

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+documentation_complete: true
+
+prodtype: rhel8,fedora,ol8
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml]:
```suggestion
prodtype: rhel6,rhel7,rhel8,fedora,ol8
```

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    Legitimate character and block devices should exist only in
+    the <tt>/dev</tt> directory on the root partition or within chroot
+    jails built for system services.
+    {{{ describe_mount(option="nodev", part="/var/log/audit") }}}
+
+rationale: |-
+    The only legitimate location for device files is the <tt>/dev</tt> directory
+    located on the root partition. The only exception to this is chroot jails.
+
+severity: unknown
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml]:
```suggestion
severity: medium
```

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+identifiers:
+    cce@rhel8: 81095-2
+
+references:
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml]: Add nist references to ``CM-7`` and ``MP-2``

CM-7 -- operate system with least privilege. Doesn't need nodev to operate

MP-2  -- multi media policy around removable drives

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    {{{ describe_mount(option="nodev", part="/var/log/audit") }}}
+
+rationale: |-
+    The only legitimate location for device files is the <tt>/dev</tt> directory
+    located on the root partition. The only exception to this is chroot jails.
+
+severity: unknown
+
+identifiers:
+    cce@rhel8: 81095-2
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nodev/rule.yml]: Please add RHEL 6 and RHEL 7 CCEs

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    Allowing users to execute binaries from world-writable directories
+    such as <tt>/var/log/audit</tt> should never be necessary in normal operation and
+    can expose the system to potential compromise.
+
+severity: unknown
+
+identifiers:
+    cce@rhel8: 82975-4
+
+references:
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml]: NIST reference to ``CM-7``

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+documentation_complete: true
+
+prodtype: rhel8,fedora,ol8
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml]:
```suggestion
prodtype: rhel6,rhel7,rhel8,fedora,ol8
```

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+rationale: |-
+    Allowing users to execute binaries from world-writable directories
+    such as <tt>/var/log/audit</tt> should never be necessary in normal operation and
+    can expose the system to potential compromise.
+
+severity: unknown
+
+identifiers:
+    cce@rhel8: 82975-4
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml]: rhel6 and rhel7 CCE

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    The <tt>noexec</tt> mount option can be used to prevent binaries
+    from being executed out of <tt>/var/log/audit</tt>.
+    {{{ describe_mount(option="noexec", part="/var/log/audit") }}}
+
+rationale: |-
+    Allowing users to execute binaries from world-writable directories
+    such as <tt>/var/log/audit</tt> should never be necessary in normal operation and
+    can expose the system to potential compromise.
+
+severity: unknown
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_noexec/rule.yml]:
```suggestion
severity: medium
```

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nosuid/rule.yml [hunk 10]
```diff
@@ -0,0 +1,24 @@
+documentation_complete: true
+
+prodtype: rhel8,fedora,ol8
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nosuid/rule.yml]:
```suggestion
prodtype: rhel6,rhel7,rhel8,fedora,ol8
```

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nosuid/rule.yml [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    {{{ describe_mount(option="nosuid", part="/var/log/audit") }}}
+
+rationale: |-
+    The presence of SUID and SGID executables should be tightly controlled. Users
+    should not be able to execute SUID or SGID binaries from temporary storage partitions.
+
+severity: unknown
+
+identifiers:
+    cce@rhel8: 82921-8
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nosuid/rule.yml]: rhel6, rhel7 CCEs

File: linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nosuid/rule.yml [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    The <tt>nosuid</tt> mount option can be used to prevent
+    execution of setuid programs in <tt>/var/log/audit</tt>. The SUID and SGID permissions
+    should not be required in these world-writable directories.
+    {{{ describe_mount(option="nosuid", part="/var/log/audit") }}}
+
+rationale: |-
+    The presence of SUID and SGID executables should be tightly controlled. Users
+    should not be able to execute SUID or SGID binaries from temporary storage partitions.
+
+severity: unknown
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/partitions/mount_option_var_log_audit_nosuid/rule.yml]:
```suggestion
severity: medium
```

**Comment by adelton** [comment]
In c8a8764ada5671a901d1cff66c2042b83806e478, I'v now rebased on master, plus I've:
- fixed the description of the purpose of these directories;
- added the medium severity;
- added the rhel8/templates/csv/mount_options.csv entries;
- added references `nist: CM-7` and `nist-csf: PR.IP-1,PR.PT-2,PR.PT-3` to all, plus `nist: MP-2` to the nodev ones.

I did not add the RHEL 6 and RHEL 7 stuff. Let's backport it there once the RHEL 8 stuff is done and added to the ospp profiles.

**Comment by yuumasato** [comment]
LGTM.
@adelton Please, revolve the conflicts.

**Comment by adelton** [comment]
Rebased on master -> 5520796fb915300a144d61aa4a7072e0528ee1c3.

**Comment by yuumasato** [comment]
@shawndwells We can add these rules to RHEL6 and RHEL7 in later PR.
Ack, thank you @adelton.

**Pull request merged**